bridges/otelzap/encoder.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
+
+import (
+	"time"
+
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+var _ zapcore.ObjectEncoder = (*objectEncoder)(nil)
+
+// objectEncoder implements zapcore.ObjectEncoder.
+// It encodes given fields to OTel key-values.
+type objectEncoder struct {
+	kv []log.KeyValue
+}
+
+func newObjectEncoder(len int) *objectEncoder {
+	keyval := make([]log.KeyValue, 0, len)
+
+	return &objectEncoder{
+		kv: keyval,
+	}
+}
+
+// TODO
+// Converts Array to log.Slice using ArrayEncoder.
+func (m *objectEncoder) AddArray(key string, v zapcore.ArrayMarshaler) error {
```

Comment:
From https://go.dev/wiki/CodeReviewComments#comment-sentences:

> Comments should begin with the name of the thing being described 

The same comment applies to documentation of all methods.

Reply:
added documentation comments for primitive types that we are sure. Will add for the rest when implementing them.